servicetalk-opentelemetry/src/test/java/io/servicetalk/opentelemetry/HeadersPropagatorGetterTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicetalk.opentelemetry;
+
+import io.servicetalk.http.api.HttpHeaders;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+
+class HeadersPropagatorGetterTest {
+
+    @Test
+    void shouldGetAllKeys() {
+
+        final HeadersPropagatorGetter getter = new HeadersPropagatorGetter();
+        HttpHeaders carrier = mock(HttpHeaders.class);
+        Set<CharSequence> set = new HashSet<>();
```

Comment:
Instead of using a mock backed by `HashSet`, consider testing with real implementations if `HttpHeaders` using `DefaultHttpHeadersFactory`.